### PR TITLE
[PROD-4557] Refactoring and addition of kafka batch handler

### PR DESCRIPTION
**Pull request by klesgidis**
## Summary
* Refactors base kafka handler
* Exports a new handler called `BaseKafkaBatchHandler` that supports batch processing for messages

### CODE REVIEW

File: test/initializers/kafka/base-kafka-handler.test.ts [hunk 1]
```diff
@@ -11,31 +11,54 @@ const logger = {
...
+    handleBatchStub = sandbox.stub();
+    heartBeatStub = sandbox.stub().resolves();
     runStub = sandbox.stub();
     subscribeStub = sandbox.stub();
     consumeStub = {
       subscribe: subscribeStub,
       run: runStub.callsFake(
-        async ({ eachMessage: fn }) =>
-          await fn({
+        async ({ eachMessage: fn, eachBatch: fnBatch }) => {
```
Comment (npetridis) [inline test/initializers/kafka/base-kafka-handler.test.ts:37]: 🥇

Comment (klesgidis) [reply test/initializers/kafka/base-kafka-handler.test.ts:37]: Re mas douleueis?

File: package-lock.json [hunk 2]
```diff
@@ -1306,7 +1306,7 @@
     },
     "buffer-equal-constant-time": {
       "version": "1.0.1",
-      "resolved": "https://registry.npmjs.org/buffer-equal-constant-time/-/buffer-equal-constant-time-1.0.1.tgz",
+      "resolved": "https://workable.jfrog.io/workable/api/npm/npm/buffer-equal-constant-time/-/buffer-equal-constant-time-1.0.1.tgz",
```
Comment (kzacharakis) [inline package-lock.json:1309]: ❓ Was this intentionally changed?

**Review by nikosd23 — APPROVED** [review]

**Review by kzacharakis — APPROVED** [review]
LGTM  
It might also worth enhancing the docs with some examples how clients can leverage the Batch Handler

File: package-lock.json (hunk 2 above)
Comment (klesgidis) [reply package-lock.json:1309]: Hm.. Since this is a public open-source project I should probably revert this

**Comment by klesgidis** [comment]
> LGTM
> It might also worth enhancing the docs with some examples how clients can leverage the Batch Handler

Yes, I forgot about that. Will do..

**Pull request merged**